Add plugin: Search In Canvas

Quorafind opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

https://github.com/Quorafind/Obsidian-Search-In-Canvas

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [ ] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["minAppVersion": "0.15.0",](https://github.com/Quorafind/Obsidian-Search-In-Canvas/blob/be2e51ba2946748c73bfa79f19ebba6061681ea6/manifest.json#L5)
Canvas was not available in that version.

[import { remote } from "electron";](https://github.com/Quorafind/Obsidian-Search-In-Canvas/blob/be2e51ba2946748c73bfa79f19ebba6061681ea6/src/searchPanel.ts#L16)
This import is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

Quorafind commented:
@joethei 

- Fixed the issue 1, I set the minAppVersion as `1.2.0` so it should support most function i used in this plugin.
- Fixed the issue 2, I add a Plateform.isDesktop to check if it is current on the desktop and then require remote package dynamically

Quorafind commented:
Btw, sorry for forgetting to add the link: https://github.com/Quorafind/Obsidian-Search-In-Canvas
